src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -723,12 +745,14 @@ def modify_job_refresh(self: "JobManager", job_ids: list[str], update_refresh: b
             self._running_jobs[job_id]["refresh"] = update_refresh
 
     def update_batch_job(self: "JobManager", batch_id: str) -> list[str]:
-        """Update a batch job and create child jobs if necessary"""
+        """Update a batch job and create child jobs if necessary."""
         batch_job = self.get_job(batch_id)
         if not batch_job.batch_job:
             raise JobRequestException(JOB_NOT_BATCH_ERR, batch_id)
 
-        child_ids = batch_job.child_jobs
+        child_ids: list = batch_job.child_jobs
+        if not child_ids:
+            return [batch_id]
```

Comment:
shortcut - if there are no child ids, no need to go through all this code that deals with them. Just return the batch ID